### ISSUE TITLE
Reg:- How to do different types of color and size for a product.

### ISSUE THREAD

**Issue by Ramesh-Palanivel** [tryshoppe/shoppe#228]
I used this gem my new rails application for garment product, we want to implement a product having 3 different color and 3 different size for a garment, here how to associate this relationship to a product model.

Thanks in advance

**Comment by deanperry** [comment]
You can use product attributes which are on the product page in the admin area. Is this what you're looking for?

**Comment by Ramesh-Palanivel** [comment]
Yes, I thing so , you are right.  But A Product having many colors and sizes , If i am using product attributes i need to create a product for each color and size. here we will create same product many time , there fore Product name is not uniq , Example : Product : T-shirt , Color: red , green , blue Size : S , M , XL . In this case i can not use product attributes fields. 

Please suggest to me.
Thanks in advance

**Comment by ikhakoo** [comment]
@Ramesh-Palanivel 

I had the same issue.  What I did was create Each product + color and then made variants for sizes.  You end up having lots of products but its the only true way to track inventory if you think about it. 

If you need some helps I suggest you check https://github.com/ikhakoo/ap 

Go to the seeds file and see how I use basic do loops to iterate through hashes of items with multiple colors and sizes.

If you need any further help, let me know.

**Comment by uaqben** [comment]
Hi, I also need to list `sizes` for garments, and manage the stock levels according to `sizes`.

I've looked into using variants, but it does feels overkill/cumbersome to go that route.

A variant allows/demands a different name/permalink/SKU/price/cost-price... Whereas, when dealing with clothes sizes, the price of the product does not change because of the size.

At least that's my experience. And if one does need different prices per different `size`, one can resort to creating `variants` to handle such special cases. 

Do you think `shoppe` could consider adding a _default_ `size` and `color` field/attribute to products?

Maybe existing product `attributes` can be improved to allow for them to be _trackable_ (when desired) in the inventory..?

Would that be a good way to go about this?

**Comment by ikhakoo** [comment]
Clone my Shoppe repo. I edited it to include sizes and colors.  It's called
Shoppe-1.0.5-AP or something like that.

**Comment by uaqben** [comment]
@ikhakoo thanks ;)

I checked your repo, but I would prefer if such size/color handling+tracking be natively managed by the `shoppe` gem. Do  you know if that is being considered?
